Products/GenericSetup/registry.py

```diff
@@ -552,12 +559,11 @@ def listForbiddenTools(self):
     def addForbiddenTool(self, tool_id):
         """ See IToolsetRegistry.
         """
+        self._required.pop(tool_id, None)
+
```

Comment:
`self._forbidden` is a list and `self._required` is a dict, so they need different handling.
Well, instead of `pop` I can use `del`, so you don't start wondering why something gets popped and not used.
Done.